script/utils/CurrentConfigCheck.s.sol

```diff
@@ -0,0 +1,570 @@
+        _verifyInitializationParams();
+
+        // bytes memory data = abi.encodeWithSelector(ProxyAdmin.changeProxyAdmin.selector, address(bEIGEN), address(beigenTokenProxyAdmin));
+        // bytes memory data = abi.encodeWithSignature("swapOwner(address,address,address)",
+        //     0xCb8d2f9e55Bc7B1FA9d089f9aC80C583D2BDD5F7,
+        //     0xcF19CE0561052a7A7Ff21156730285997B350A7D,
+        //     0xFddd03C169E3FD9Ea4a9548dDC4BedC6502FE239
+        // );
+        // bytes memory callToExecutor = encodeForExecutor({
+        //     from: communityMultisig,
+        //     to: address(executorMultisig),
+        //     value: 0,
+        //     data: data,
+        //     operation: ISafe.Operation.Call 
+        // });
+
+        // vm.prank(communityMultisig);
+        // (bool success, /*bytes memory returndata*/) = executorMultisig.call(callToExecutor);
+        // require(success, "call to executorMultisig failed");
+
+        // vm.startPrank(foundationMultisig);
+        // Ownable(address(bEIGEN)).transferOwnership(address(eigenTokenTimelockController));
+        // Ownable(address(EIGEN)).transferOwnership(address(eigenTokenTimelockController));
+        // vm.stopPrank();
+
+        checkGovernanceConfiguration_Current();
+
+        // simulateProtocolCouncilUpgrade(networkName);
+    }
+
+    // check governance configuration
+    function checkGovernanceConfiguration_Current() public {
+        assertEq(eigenLayerProxyAdmin.owner(), executorMultisig,
+            "eigenLayerProxyAdmin.owner() != executorMultisig");
+        assertEq(delegationManager.owner(), executorMultisig,
+            "delegationManager.owner() != executorMultisig");
+        assertEq(strategyManager.owner(), executorMultisig,
+            "strategyManager.owner() != executorMultisig");
+        assertEq(strategyManager.strategyWhitelister(), address(strategyFactory),
+            "strategyManager.strategyWhitelister() != address(strategyFactory)");
+        assertEq(strategyFactory.owner(), operationsMultisig,
+            "strategyFactory.owner() != operationsMultisig");
+        assertEq(avsDirectory.owner(), executorMultisig,
+            "avsDirectory.owner() != executorMultisig");
+        assertEq(rewardsCoordinator.owner(), operationsMultisig,
+            "rewardsCoordinator.owner() != operationsMultisig");
+        assertEq(eigenLayerPauserReg.unpauser(), executorMultisig,
+            "eigenLayerPauserReg.unpauser() != operationsMultisig");
+        require(eigenLayerPauserReg.isPauser(operationsMultisig),
+            "operationsMultisig does not have pausing permissions");
+        require(eigenLayerPauserReg.isPauser(executorMultisig),
+            "executorMultisig does not have pausing permissions");
+        require(eigenLayerPauserReg.isPauser(pauserMultisig),
+            "pauserMultisig does not have pausing permissions");
+
+        (bool success, bytes memory returndata) = timelock.staticcall(abi.encodeWithSignature("admin()"));
```

Comment:
why not cast to `Timelock`?

Reply:
thought we didn't have the interface in here 🤷 
probably should do that though

Reply:
switched this in https://github.com/Layr-Labs/eigenlayer-contracts/pull/794/commits/d5844681cdfcc2a90b80a80ac97d440e1bf39736